services/common/src/components/help/HelpGuide.tsx

```diff
@@ -0,0 +1,203 @@
+      dispatch(deleteHelp({ helpKey, help_guid })).then(() => {
+        setIsEditMode(false);
+      })
+    );
+  };
+
+  const triggerSubmit = () => {
+    dispatch(submit(FORM.EDIT_HELP_GUIDE));
+  };
+
+  const handleSaveGuide = (values) => {
+    const page_tab = values.page_tab ?? HELP_GUIDE_ALL_TABS;
+    const data = { ...values, page_tab, system, help_key: helpKey };
+
+    const payload = { helpKey, data };
+    const forNewTab =
+      helpGuide.page_tab === HELP_GUIDE_ALL_TABS && values.page_tab !== HELP_GUIDE_ALL_TABS;
+    const saveFunction =
+      defaultGuide || forNewTab
+        ? () => dispatch(createHelp(payload))
+        : () => dispatch(updateHelp(payload));
+
+    saveFunction().then((resp) => {
+      if (resp.payload) {
+        setIsEditMode(false);
+      }
+    });
+  };
+
+  const getFooterButtons = () => {
+    const editButton = (
+      <Button type="primary" onClick={() => setIsEditMode(true)}>
+        {hasHelpGuide ? "Edit Help Guide" : "Create Help Guide"}
+      </Button>
+    );
+    const submitButton = (
+      <Button type="primary" onClick={triggerSubmit}>
+        Publish Help Guide
+      </Button>
+    );
+    const deleteButton = (
+      <Button disabled={!hasHelpGuide || defaultGuide} onClick={handleDeleteGuide}>
+        Delete this Guide
+      </Button>
+    );
+    const buttons = isEditMode ? (
+      <>
+        <Col>{deleteButton}</Col>
+        <Col>{submitButton}</Col>
+      </>
+    ) : (
+      <Col>{editButton}</Col>
+    );
+    const justify = isEditMode ? "space-between" : "end";
+    return <Row justify={justify}>{buttons}</Row>;
+  };
+  const initialValues =
+    hasHelpGuide && !defaultGuide ? helpGuide : { help_key: helpKey, page_tab: pageTab };
+  const mainContent = isEditMode ? (
+    <HelpGuideForm
+      initialValues={initialValues}
+      handleSaveGuide={handleSaveGuide}
+      pageTab={pageTab}
+    />
+  ) : (
+    parse(DOMPurify.sanitize(content))
+  );
+
+  return (
+    <>
+      <Button data-testid="help-open" onClick={showDrawer}>
+        ?
+      </Button>
+      <Drawer
+        placement="right"
+        className="help-guide-drawer"
+        onClose={hideDrawer}
+        open={open}
+        title="Help"
+        footer={canEditHelp && getFooterButtons()}
+        destroyOnClose
+      >
+        <div data-testid="help-content">
+          {isEditMode && (
+            <Alert
+              message="Publish Help Guide"
+              showIcon
+              type="warning"
+              description="Content published here will be visible to MineSpace users. Ensure the language is appropriate and avoid sharing confidential information."
+            />
+          )}
+          <Typography.Title level={2} data-testid="help-title">
+            {title} Help Guide
+          </Typography.Title>
+          {isLoaded ? mainContent : <Loading />}
+        </div>
+      </Drawer>
+    </>
+  );
+};
+
+const HelpGuide: FC = () => {
```

Comment:
I guess this is just personal preference, but I generally prefer to have functional components in their own files where possible.  I'll leave it up to you though.

Reply:
I think I'll leave it this time... The sub-component I only turned into an export so that I'd be able to test it easier!